src/test/kotlin/io/sip3/salto/ce/socket/ManagementSocketTest.kt

```diff
@@ -0,0 +1,154 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sip3.salto.ce.socket
+
+import io.sip3.commons.domain.SdpSession
+import io.sip3.commons.vertx.test.VertxTest
+import io.sip3.salto.ce.MongoExtension
+import io.sip3.salto.ce.RoutesCE
+import io.sip3.salto.ce.USE_LOCAL_CODEC
+import io.vertx.core.json.JsonObject
+import io.vertx.ext.mongo.MongoClient
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import java.net.ServerSocket
+import java.util.*
+
+@ExtendWith(MongoExtension::class)
+class ManagementSocketTest : VertxTest() {
+
+    companion object {
+
+        private val host = JsonObject().apply {
+            put("name", "sbc1")
+            put("sip", arrayListOf("10.10.10.10", "10.10.20.10:5060"))
+        }
+    }
+
+    private lateinit var config: JsonObject
+    private var localPort = -1
+    private var remotePort = -1
+    private val registerMessage = JsonObject().apply {
+        put("type", ManagementSocket.TYPE_REGISTER)
+        put("payload", JsonObject().apply {
+            put("name", UUID.randomUUID().toString())
+            put("host", host)
+        })
+    }
+
+    @BeforeEach
+    fun init() {
+        val localSocket = ServerSocket(0)
+        localPort = localSocket.localPort
+        val remoteSocket = ServerSocket(0)
+        remotePort = remoteSocket.localPort
+        localSocket.close()
+        remoteSocket.close()
+
+        config = JsonObject().apply {
+            put("mongo", JsonObject().apply {
+                put("uri", "mongodb://${MongoExtension.HOST}:${MongoExtension.PORT}")
+                put("db", "sip3")
+                put("bulk-size", 1)
+            })
+            put("management", JsonObject().apply {
+                put("uri", "udp://127.0.0.1:$localPort")
+                put("expiration-timeout", 1500L)
+                put("expiration-delay", 800L)
+            })
+        }
+    }
+
+    @Test
+    fun `Receive register from remote host with host information`() {
+        runTest(
+                deploy = {
+                    vertx.deployTestVerticle(ManagementSocket::class, config)
+                },
+                execute = {
+                    vertx.createDatagramSocket().send(registerMessage.toBuffer(), localPort, "127.0.0.1") {}
+                },
+                assert = {
+                    vertx.setTimer(1000L) {
+                        val mongo = MongoClient.createShared(vertx, JsonObject().apply {
+                            put("connection_string", "mongodb://${MongoExtension.HOST}:${MongoExtension.PORT}")
+                            put("db_name", "sip3")
+                        })
+                        vertx.setPeriodic(100) {
+                            mongo.find("hosts", JsonObject()) { asr ->
```

Comment:
let's use `mongo.findAwait`

Reply:
Also we can call `findOne`, checking for HOST itself. Assertion below will be way smaller